Docs: PyPI Setup & Deploy #67 (#80)
AmateursLeague/sneaky-package: PyPi_Setup.md

# PyPi setup 

## 1. Introduction

> This documentation provides a comprehensive guide on how to set up and deploy a Python package to the Python Package Index (PyPI). The package is designed to facilitate the sharing of Python projects, enabling users to install it easily using pip.

Reviewer: I feel like this can be better, it doesn't sound that good with that `This documentation` it should be something in the first person I feel. CC @this-is-yaash

### Purpose

> The objectives of this guide are to help users:

Reviewer: Can you change this to first person, idk is sounds a little wired while mentoning users in the docs, it should be something like 
```suggestion
The objectives of this guide is to help you with:
```

## 3. Installations

To prepare your environment, upgrade pip, setuptools, and wheel:

```bash
sudo python -m pip install --upgrade pip setuptools wheel
```

Next, install `tqdm` and `Twine`:

```bash
sudo python -m pip install tqdm
sudo python -m pip install --user --upgrade twine
```

These tools are essential for building and uploading your package.

## 4. Project Structure

Here’s an example of a basic project structure for a Python package:

```
my_package/
│
├── setup.py              # Setup script for packaging
├── README.md             # Package documentation
├── LICENSE               # License for the package
├── my_package/           # Main package directory
│   ├── __init__.py       # Required for the package
│   └── module.py         # Your module code
└── tests/                # Unit tests for the package
    └── test_module.py    # Example test file
```

### Explanation of Key Files

- **setup.py**: Contains package information and dependencies.
- **README.md**: Provides information about the package.
- **my_package/**: Directory containing the actual Python code.
- **tests/**: Directory for unit testing your package.

> 

Reviewer 2: Use the actual repository project structure.
It seems totally unrelated! Doesn't make sense!

Reviewer 2: Use same naming conventions as well, 
But at the end, add a saying that feel free change the names

and inside the `stash` dir contains the features, mention that these filenames are aliases. feel free to change it to meet your needs!

## 5. Building the Package

### Step-by-Step Instructions

1. Create a `setup.py` file in the root directory with the following content:

```python
from setuptools import setup, find_packages

setup(
    name='your_package_name',
    version='0.1',
    packages=find_packages(),
    install_requires=[
        # Add any package dependencies here
    ],
    # You can also add metadata like author, license, etc.
)

> ```

Reviewer 2: Use the `setup.py` in this repo for reference. Copy-pasting is also enf!

**Note:** Ensure that your package's structure is correct and that the `setup.py` file is properly configured.

2. Build the Package by running:

```bash
python setup.py sdist bdist_wheel
```

This command will create a `dist/` directory containing `.tar.gz` and `.whl` files.

## 6. Uploading to PyPI

### Detailed Steps for Deployment

1. **Create a PyPI Account**:
   - Go to PyPI and create an account if you don’t have one.
   - After logging in, generate an API token via your account settings.

2. **Upload the Package Using Twine**:

>    - Use your API token for authentication:

Reviewer 2: Elaborate with steps, it's going to be tedious process if documented with lil info!
